actionview/lib/action_view/path_set.rb

```diff
@@ -44,28 +44,38 @@ def #{method}(*args)
       METHOD
     end
 
-    def find(*args)
-      find_all(*args).first || raise(MissingTemplate.new(self, *args))
+    def find(path, prefixes, partial, details, details_key, locals)
+      search_combinations(prefixes) do |resolver, prefix|
+        template = resolver.find_all(path, prefix, partial, details, details_key, locals).first
```

Comment:
Any reason why we `find_all` was moved out of the private method? It seems we could avoid that duplication with no cost.

Reply:
So eventually I'm expecting these implementations to become more different. IMO the current `find_all` has the wrong behaviour (it doesn't return _all_ the templates, just _all_ the templates from the first path/prefix combination it finds). I also intend to make `exists?` faster by skipping the "locals" lookup (unnecessary to determine a template's existence), and I figured I might as well skip the two extra method calls here.

But maybe I'm getting ahead of myself and can do all that in a future refactor 😅.